test/getConfig.spec.js

```diff
@@ -0,0 +1,242 @@
+        concurrent: true
+      })
+    )
+
+    expect(
+      getConfig({
+        concurrent: false
+      })
+    ).toEqual(
+      expect.objectContaining({
+        concurrent: false
+      })
+    )
+
+    expect(
+      getConfig({
+        concurrent: true
+      })
+    ).toEqual(
+      expect.objectContaining({
+        concurrent: true
+      })
+    )
+  })
+
+  it('should set renderer based on verbose key', () => {
+    expect(getConfig({})).toEqual(
+      expect.objectContaining({
+        renderer: 'update'
+      })
+    )
+
+    expect(
+      getConfig({
+        '*.js': ['eslint', 'git add']
+      })
+    ).toEqual(
+      expect.objectContaining({
+        renderer: 'update'
+      })
+    )
+
+    expect(
+      getConfig({
+        verbose: false
+      })
+    ).toEqual(
+      expect.objectContaining({
+        renderer: 'update'
+      })
+    )
+
+    expect(
+      getConfig({
+        verbose: true
+      })
+    ).toEqual(
+      expect.objectContaining({
+        renderer: 'verbose'
+      })
+    )
+  })
+
+  it('should set gitDir', () => {
+    expect(getConfig({})).toEqual(
+      expect.objectContaining({
+        gitDir: '.'
+      })
+    )
+
+    expect(
+      getConfig({
+        gitDir: '../path'
+      })
+    ).toEqual(
+      expect.objectContaining({
+        gitDir: '../path'
+      })
+    )
+  })
+
+  it('should set linters', () => {
+    expect(getConfig()).toEqual(
+      expect.objectContaining({
+        linters: {}
+      })
+    )
+
+    expect(getConfig({})).toEqual(
+      expect.objectContaining({
+        linters: {}
+      })
+    )
+
+    expect(
+      getConfig({
+        '*.js': 'eslint'
+      })
+    ).toEqual(
+      expect.objectContaining({
+        linters: {
+          '*.js': 'eslint'
+        }
+      })
+    )
+
+    expect(
+      getConfig({
+        linters: {
+          '*.js': ['eslint --fix', 'git add'],
+          '.*rc': 'jsonlint'
+        }
+      })
+    ).toMatchSnapshot()
+  })
+
+  it('should deeply merge configs', () => {
+    expect(
+      getConfig({
+        globOptions: {
+          nocase: true
+        }
+      })
+    ).toEqual(
+      expect.objectContaining({
+        globOptions: {
+          nocase: true,
+          matchBase: true,
+          dot: true
+        }
+      })
+    )
+  })
+
+  it('should not add plain linters object to the full config', () => {
+    expect(
+      getConfig({
+        '*.js': 'eslint'
+      })
+    ).not.toEqual(
+      expect.objectContaining({
+        '*.js': 'eslint'
+      })
+    )
+  })
+
+  it('should not change config if the whole config was passed', () => {
+    const src = {
+      concurrent: false,
+      chunkSize: 2,
+      gitDir: '/to',
+      globOptions: {
+        matchBase: false,
+        dot: true
+      },
+      linters: {
+        '*.js': 'eslint'
+      },
+      subTaskConcurrency: 10,
+      renderer: 'custom',
+      verbose: true
+    }
+    expect(getConfig(src)).toEqual(src)
```

Comment:
If `src` was being mutated by the `getConfig`, this test would still pass. How about we use [`lodash#cloneDeep`](https://lodash.com/docs/4.17.4#cloneDeep) to pass a copy of `src` to `getConfig`?

Reply:
You're right. Nice catch!